source/common/http/path_utility.cc

```diff
@@ -52,5 +54,20 @@ bool PathUtil::canonicalPath(HeaderEntry& path_header) {
   return true;
 }
 
+void PathUtil::mergeSlashes(HeaderEntry& path_header) {
```

Comment:
I'm a bit twitchy about full qualified URLs here.  Granted, Envoy currently handles fully qualified urls in the H1 codec by splitting them up, so if the incoming request is GET http://foo.com//bar we'll have :authority foo.com and :path /bar so this is fine today (we won't end up with http:/foo.com/bar).  

It might be worth an assert that the path is relative, just in case someone does some lua or use defined headers  trying to take advantage of Path == url in firstline and having their absolute URL messed up.